### PR TITLE
refactor: slice finder UI and add continuous columns

### PR DESCRIPTION
**Pull request by stevenyh3**
This PR introduces continuous columns to the available search column options, refactors the user interface, and provide slice naming feature prior to creating the slice using the slice finder.

### CODE REVIEW

**Review by cabreraalex — APPROVED** [review]

File: frontend/src/metadata/popups/SliceFinderPopup.svelte [hunk 1]
```diff
@@ -172,85 +182,132 @@
...
 			<div>
-				<div class="options-header">Order By</div>
+				<div style="display:flex">
+					<div class="options-header">Order By</div>
+					<div
+						class="information-tooltip"
+						style="margin-top: 3px;"
+						use:tooltip={{
+							content:
+								"Ordered by cumulative slice score obtained through the sliceline function",
```
Comment (cabreraalex) [inline frontend/src/metadata/popups/SliceFinderPopup.svelte:214]: maybe `Order by slice score, a combination of size and metric`

File: zeno/processing/slice_finder.py [hunk 2]
```diff
@@ -17,37 +29,40 @@ def slice_finder(df, req: SliceFinderRequest):
...
     """
+    cont_search_cols = [str(col) for col in req.search_columns_cont]
+    not_cont_search_cols = [str(col) for col in req.search_columns]
+
+    cont_df = cont_cols_df(df[cont_search_cols].dropna(), cont_search_cols)
 
-    updated_df = df[
-        list(set([str(col) for col in req.columns] + [str(req.metric_column)]))
-    ].dropna()
+    uniq_cols = set(not_cont_search_cols + [str(req.metric_column)])
```
Comment (cabreraalex) [inline zeno/processing/slice_finder.py:37]: `unique_cols`

**Review by cabreraalex — CHANGES_REQUESTED** [review]

File: frontend/src/metadata/cells/SliceFinderCell.svelte [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+		ZenoService.createNewSlice(slice).then(() => {
+			slices.update((s) => {
+				s.set(newSliceName, slice);
+				return s;
+			});
+		});
+		createdSliceName = newSliceName;
+		showSliceName = false;
+		created = true;
+	}
```
[the remark this answers is not in the document]
Comment (stevenyh3) [reply frontend/src/metadata/cells/SliceFinderCell.svelte:41]: done

File: frontend/src/metadata/cells/SliceFinderCell.svelte [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+	function removeSlice() {
+		slices.update((s) => {
+			s.delete(createdSliceName);
+			return s;
+		});
+		deleteSlice(createdSliceName);
```
[the remark this answers is not in the document]
Comment (stevenyh3) [reply frontend/src/metadata/cells/SliceFinderCell.svelte:45]: yeah, this makes sense. I've modify the deleteSlice function in api/slice.ts.

File: frontend/src/metadata/cells/SliceFinderCell.svelte [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+			on:keydown={submit}
+			use:clickOutside
+			on:click_outside={() => (showSliceName = false)}>
+			<Paper elevation={7}>
+				<Content style="display:flex; flex-direction:column">
+					<Textfield
+						bind:value={newSliceName}
+						label="Slice Name"
+						bind:this={input} />
+					{#if $slices.has(newSliceName)}
```
[the remark this answers is not in the document]
Comment (stevenyh3) [reply frontend/src/metadata/cells/SliceFinderCell.svelte:103]: I don't think so, as the error message "Slice already exists" will show when slice name input is empty.

File: frontend/src/metadata/cells/SliceFinderCell.svelte [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+					<path fill="#6a1b9a" d={mdiPlus} />
+				</Icon>
+			</IconButton>
+		{/if}
+	</div>
+	{#if showSliceName}
+		<div
+			id="slice-name"
+			on:keydown={submit}
+			use:clickOutside
```
[the remark this answers is not in the document]
Comment (stevenyh3) [reply frontend/src/metadata/cells/SliceFinderCell.svelte:94]: done

File: frontend/src/metadata/cells/SliceFinderCell.svelte [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+						bind:this={input} />
+					{#if $slices.has(newSliceName)}
+						<p style="margin-right: 10px; color: red">Slice already exists!</p>
+					{/if}
+					<div class="buttons">
+						<Button
+							style="margin-left: 10px;"
+							variant="outlined"
+							on:click={() => (showSliceName = false)}
+							>Cancel
```
[the remark this answers is not in the document]
Comment (stevenyh3) [reply frontend/src/metadata/cells/SliceFinderCell.svelte]: done

File: frontend/src/metadata/cells/SliceFinderCell.svelte (hunk 5 above)
[the remark this answers is not in the document]
Comment (cabreraalex) [reply frontend/src/metadata/cells/SliceFinderCell.svelte:103]: Sounds good!

File: frontend/src/metadata/popups/SliceFinderPopup.svelte [hunk 8]
```diff
@@ -1,57 +1,54 @@
...
-					d.columnType === ZenoColumnType.POSTDISTILL) &&
-					d.model === $model))
+			notEmbedUniqCols.includes(d)
 	);
 	let metricColumn = metricColumns.length > 0 ? metricColumns[0] : null;
 
 	let alphas = ["0.5", "0.75", "0.9", "0.95", "0.99", "0.999"];
-	let alphaIdx = 3;
+	let alphaIdx = 4;
 	let minimumSupps = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "10"];
```
[the remark this answers is not in the document]
Comment (stevenyh3) [reply frontend/src/metadata/popups/SliceFinderPopup.svelte:52]: Actually I didn't see any difference when changing the min_support variable, even passing floating numbers like 0.5, 0.25... So Idk which paradigm might be better...

[the remark this answers is not in the document]
Comment (stevenyh3) [reply frontend/src/metadata/popups/SliceFinderPopup.svelte]: Replace min_support with max lattice size with default 4 and options being [1,2,3,4,5,6]

File: frontend/src/metadata/popups/SliceFinderPopup.svelte [hunk 9]
```diff
@@ -66,37 +63,34 @@
...
-			slices.update((s) => {
-				s.set(slice.sliceName, slice);
-				return s;
-			});
-			showNewSlice.set(false);
-			sliceToEdit.set(null);
-		});
-	}
-
 	export async function activateSliceFinder() {
```
[the remark this answers is not in the document]
Comment (stevenyh3) [reply frontend/src/metadata/popups/SliceFinderPopup.svelte]: done

File: frontend/src/metadata/popups/SliceFinderPopup.svelte (hunk 1 above)
Comment (stevenyh3) [reply frontend/src/metadata/popups/SliceFinderPopup.svelte]: done

File: frontend/src/metadata/popups/SliceFinderPopup.svelte [hunk 10]
```diff
@@ -172,85 +182,132 @@
...
 				</div>
 			</div>
+			<div class="generation" style="margin-bottom:0px;">
+				<h4 style="margin-bottom:0px;">Filter Predicates</h4>
+				<h4 style="margin-bottom:0px;">Slice Average Metrics</h4>
+			</div>
+		{:else}
+			<div id="initial">
+				<span class="intial-text" style="font-weight: bold"
+					>Click below to find slices with low performance!</span>
```
[the remark this answers is not in the document]
Comment (stevenyh3) [reply frontend/src/metadata/popups/SliceFinderPopup.svelte]: done

File: zeno/classes/slice_finder.py [hunk 11]
```diff
@@ -6,7 +6,8 @@
 
 class SliceFinderRequest(CamelModel):
     metric_column: ZenoColumn
-    columns: List[ZenoColumn]
+    search_columns_cont: List[ZenoColumn]
```
[the remark this answers is not in the document]
Comment (stevenyh3) [reply zeno/classes/slice_finder.py]: Changed to a simpler API.

File: zeno/processing/slice_finder.py [hunk 12]
```diff
@@ -2,12 +2,24 @@
...
 
 import numpy as np
+import pandas as pd
 from sliceline.slicefinder import Slicefinder
 
 from zeno.classes.slice import FilterPredicate, FilterPredicateGroup, Slice
 from zeno.classes.slice_finder import SliceFinderRequest, SliceFinderReturn
 
 
+# encode continuous value to range bins categorical columns
```
[the remark this answers is not in the document]
Comment (stevenyh3) [reply zeno/processing/slice_finder.py]: done

File: zeno/processing/slice_finder.py (hunk 2 above)
Comment (stevenyh3) [reply zeno/processing/slice_finder.py]: done

**Review by cabreraalex — APPROVED** [review]

**Pull request merged**